server/src/main/java/io/crate/expression/operator/any/AnyEqOperator.java

```diff
@@ -58,11 +58,11 @@ boolean matches(Object probe, Object candidate) {
     }
 
     @Override
-    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, @NotNull List<?> nonNullValues, Context context) {
+    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, @NotNull List<?> nonNullValues, int arrayLiteralDimensions, Context context) {
         String columnName = probe.storageIdent();
         DataType<?> type = probe.valueType();
         DataType<?> innerType = ArrayType.unnest(type);
-        if (ArrayType.dimensions(type) > 1) {
+        if (arrayLiteralDimensions > 1) {
```

Comment:
Originally assumed that the dimension of the array literal is the dimension of ref + 1 but it is not true due to automatic dimensions levelling . i.e. `[1] = any([ [ [1] ] ]) --> true`.